### PR TITLE
feat: added intermediate event change

### PR DESCRIPTION
**Pull request by truongductri01**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes  https://github.com/google/blockly/issues/7601

Intermediate Change

### CODE REVIEW

File: core/events/events_block_field_intermediate_change.ts [hunk 1]
```diff
@@ -122,6 +122,37 @@ export class BlockFieldIntermediateChange extends BlockBase {
   override isNull(): boolean {
     return this.oldValue === this.newValue;
   }
+
+  /**
+   * Run a change event.
+   *
+   * @param forward True if run forward, false if run backward (undo).
+   */
+  override run(forward: boolean) {
```
Comment (BeksOmega) [inline core/events/events_block_field_intermediate_change.ts:131]: This looks perfect!

File: tests/mocha/event_block_field_intermediate_change_test.js [hunk 2]
```diff
@@ -35,4 +35,38 @@ suite('Field Intermediate Change Event', function () {
       chai.assert.deepEqual(newEvent, origEvent);
     });
   });
+
+  suite('Change Value', function () {
```
Comment (BeksOmega) [inline tests/mocha/event_block_field_intermediate_change_test.js:39]: Thank you for adding tests :D

File: tests/mocha/event_block_field_intermediate_change_test.js [hunk 3]
```diff
@@ -35,4 +35,38 @@ suite('Field Intermediate Change Event', function () {
...
+    test("runs with forward change block's value to new value", function () {
```
Comment (BeksOmega) [inline tests/mocha/event_block_field_intermediate_change_test.js:40]: Can you change these to be of the form "this tests that X" where X becomes the title of the test?

E.g. "running forward changes the block's value to the new value", and "running backward changes the block's value to the old value".

**Comment by truongductri01** [comment]
> Thank you for your work on this! Just a few nits then should be good to go =)

I have fixed the codes as requested. There is a conversation in the issue ticket that I wish to receive your response. I have some issues with building the code in my forked repo.

File: tests/mocha/event_block_field_intermediate_change_test.js [hunk 4]
```diff
@@ -35,4 +35,38 @@ suite('Field Intermediate Change Event', function () {
       chai.assert.deepEqual(newEvent, origEvent);
     });
   });
+
+  suite('Change Value', function () {
+    test("this tests that running forward changes the block's value to new value", function () {
```
Comment (BeksOmega) [inline tests/mocha/event_block_field_intermediate_change_test.js:40]: Can you remove the "this tests that" part of the titles?

**Review by BeksOmega — COMMENTED** [review]
One last nit then should be good to go! Thank you for making the updates :D

**Review by BeksOmega — APPROVED** [review]
This looks great! Thank you for your work on this :D

**Pull request merged**